### PR TITLE
Rust: Add `geo-traits` reader implementation

### PR DESCRIPTION
**Pull request by kylebarron**
We have a new core `geo` crate: [`geo-traits`](https://docs.rs/geo-traits/latest/geo_traits/)! This enables zero-copy vector data interchange throughout the GeoRust ecosystem.

This PR implements these traits on new-types around a `flatgeobuf::Geometry`.

Since Flatbuffers are zero-copy, this means that you could memory-map a FlatGeobuf file and pass those geometry references directly to any consumer accepting these traits.

### Change list 

- Adds a public method onto `FgbFeature` to return a struct that implements `geo_traits::GeometryTrait`.
- Since the method returns an `impl geo_traits::GeometryTrait`, none of the new structs need to be exposed publicly.
- This provides a standard way to access FlatGeobuf data with Z and M values as well.

### Questions

- How to handle FlatGeobuf files with `t` values? Those are currently omitted
- How should we handle curve types? This PR currently panics on those.

### CODE REVIEW

File: src/rust/src/properties_reader.rs
Comment (kylebarron) [inline src/rust/src/properties_reader.rs:73]: I think this is the primary remaining question in this PR before it can receive a full review: should this function return a result? Should this error on non-simple geometry types?

Comment (pka) [reply src/rust/src/properties_reader.rs:73]: I think returning a `Result` is better, since you could also have a non-supported geometry type in a GeometryCollection.

File: src/rust/src/properties_reader.rs
Comment (kylebarron) [inline src/rust/src/properties_reader.rs:45]: I learned in https://github.com/kylebarron/wkb that this `use<'_>` syntax requires Rust 1.82. https://rust-lang.github.io/rfcs/3617-precise-capturing.html

This signature would be nice to use as it lets all of `geo_trait_impl` be a private module. I don't see a current MSRV:
https://github.com/kylebarron/flatgeobuf/blob/7e0f27e50ddde197164afcb532132796c775609a/src/rust/Cargo.toml#L2-L13

Thoughts?

File: src/rust/src/properties_reader.rs
Comment (kylebarron) [inline src/rust/src/properties_reader.rs:74]: Should this return `flatgeobuf::Error` instead of `GeozeroError`? I'd think it should, however none of the `flatgeobuf::Error` variants are obvious to use:
```
pub enum Error {
    MissingMagicBytes,
    NoIndex,
    HttpClient([HttpError](https://docs.rs/http-range-client/0.8.0/x86_64-unknown-linux-gnu/http_range_client/error/enum.HttpError.html)),
    IllegalHeaderSize([usize](https://doc.rust-lang.org/nightly/std/primitive.usize.html)),
    InvalidFlatbuffer([InvalidFlatbuffer](https://docs.rs/flatbuffers/24.3.25/x86_64-unknown-linux-gnu/flatbuffers/verifier/enum.InvalidFlatbuffer.html)),
    IO([Error](https://doc.rust-lang.org/nightly/std/io/error/struct.Error.html)),
}
```

File: src/rust/src/properties_reader.rs
Comment (pka) [reply src/rust/src/properties_reader.rs:45]: Adding an MSRV 1.82 is fine for me.

File: src/rust/src/properties_reader.rs
Comment (pka) [reply src/rust/src/properties_reader.rs:74]: Yes, we should add an FGB Error for that.

**Comment by kylebarron** [comment]
Note that we may want to wait for geo-traits 0.2 before merging and releasing: https://github.com/georust/geo/pull/1263

**Review by pka — APPROVED** [review]
A usage example would be nice.

**Comment by bjornharrtell** [comment]
@kylebarron seems like very nice work (even though I seem to get less rather than more into rust, unfortunately) - I'll merge now and see if I can make it into a published crate not too far off.

File: src/rust/src/error.rs
Comment (michaelkirk) [inline src/rust/src/error.rs:13]: Note that adding a new error variant is a breaking change unless the enum is marked as non-exhaustive.

**Pull request merged**
